src/compass/utils/validate_cslc.py

```diff
@@ -0,0 +1,144 @@
+
+    if not os.path.exists(file_2):
+        print(f'ERROR secondary CSLC product not found: {file_2}')
+        return
+
+    # Extract some info from reference/secondary CSLC products
+    dataset_1 = gdal.Open(file_1, gdal.GA_ReadOnly)
+    geotransform_1 = dataset_1.GetGeoTransform()
+    nbands_1 = dataset_1.RasterCount
+
+    dataset_2 = gdal.Open(file_2, gdal.GA_ReadOnly)
+    geotransform_2 = dataset_2.GetGeoTransform()
+    nbands_2 = dataset_2.RasterCount
+
+    # Compare number of bands
+    print('Comparing CSLC number of bands ...')
+    if not nbands_1 == nbands_2:
+        print(f'ERROR Number of bands in reference CSLC {nbands_1} differs'
+              f'from number of bands {nbands_2} in secondary CSLC')
+        return
+
+    print('Comparing geo transform arrays ...')
+    if not np.array_equal(geotransform_1, geotransform_2):
+        print(f'ERROR Reference geo transform array {geotransform_1} differs'
+              f'from secondary CSLC geo transform array {geotransform_2}')
+        return
+
+    # Compare amplitude of reference and generated CSLC products
+    slc_1 = dataset_1.GetRasterBand(1).ReadAsArray()
+    slc_2 = dataset_2.GetRasterBand(1).ReadAsArray()
+
+    diff_amp = np.abs(slc_1) - np.abs(slc_2)
+    diff_pha = np.angle(slc_1 * np.conj(slc_2))
+
+    print('Check max amplitude difference between CSLC products is < 1.0e-12')
+    assert np.nanmax(diff_amp) < 1.0e-12
+    print('Check max phase difference between CSLC products is < 1.0e-12')
+    assert np.nanmax(diff_pha) < 1.0e-12
+    return
+
+
+def compare_cslc_metadata(file_1, file_2):
+    '''
+    Compare reference and generated CSLC metadata
+    '''
+
+    # Check if metadata files exists
+    if not os.path.exists(file_1):
+        print(f'ERROR reference CSLC metadata not found: {file_1}')
+        return
+
+    if not os.path.exists(file_2):
+        print(f'ERROR CSLC metadata not found: {file_2}')
+        return
+
+    # Load metadata
+    with open(file_1, 'r') as f:
+        metadata_1 = json.load(f)
+
+    with open(file_2, 'r') as f:
+        metadata_2 = json.load(f)
+
+    print('Compare number of metadata keys')
+    if not len(metadata_1.keys()) == len(metadata_2.keys()):
+        print('ERROR different number of metadata keys')
+        return
+
+    # Intersect metadata keys
+    set_1_m_2 = set(metadata_1.keys()) - set(metadata_2.keys())
+    if len(set_1_m_2) > 0:
+        print(f'Reference CSLC metadata has extra entries with keys:'
+              f'{", ".join(set_1_m_2)}.')
+        return
+    set_2_m_1 = set(metadata_2.keys()) - set(metadata_1.keys())
+    if len(set_2_m_1) > 0:
+        print(f'Secondary CSLC metadata has extra entries with keys:'
+              f'{", ".join(set_2_m_1)}.')
+
+    # Check remaining metadatakeys
+    for k1, v1 in metadata_1.items():
+        if k1 not in metadata_2.keys():
+            print(f'ERROR the metadata key {key1} in not present'
+                  f'in the secondary CSLC metadata')
```

Comment:
```suggestion
```
Redundant - the check of set difference `set_ref_minus_sec` does the same above.